metadata/src/main/java/org/apache/kafka/metadata/bootstrap/BootstrapMetadata.java

```diff
@@ -117,7 +117,7 @@ public List<ApiMessageAndVersion> records() {
     }
 
     public MetadataVersion metadataVersion() {
-        return metadataVersion;
+        return MetadataVersion.fromFeatureLevel(metadataVersionLevel);
```

Comment:
this seems a bit dangerous - right now we call this when generating activation records and in a data loss scenario/new controller scenario - wouldn't we hit the bug again?

Reply:
In a data loss scenario with `kraft.version=1` we would need to re-format the directory and re-add the voter, so there would be no more `bootstrap.metadata` file.

(If you delete some files in the raft log directory but not others, bad things may happen, of course, but that is no different than today; it breaks the Raft invariants.)

Reply:
Right. Kafka assumes that if there is data loss it is because the entire disk failed. Kafka doesn't handle error in the filesystem were some data was lost after an fsync.

Reply:
got it - and controllers don't check that the bootstrap metadata file / bootstrap MV matches? in this data loss scenario, it would work to re-format the directory with the current MV that the cluster is on before re-adding the voter?